templates/repo/clone_script.tmpl

```diff
@@ -0,0 +1,19 @@
+<script>
+	// synchronously set clone button states and urls here to avoid flickering
+	// on page load. initRepoCloneLink calls this when proto changes.
+	(window.updateCloneStates = function() {
+		const proto = localStorage.getItem('repo-clone-protocol') || 'https';
+		const btn = document.getElementById(`repo-clone-${proto}`);
+
+		if (btn) {
+			for (const el of document.getElementsByClassName('clone-url')) {
+				el[el.nodeName === 'INPUT' ? 'value' : 'textContent'] = btn.getAttribute('data-link');
+			}
+		}
```

Comment:
Here seems incorrect. If proto  = 'https' but there is only a 'ssh' method, then btn is null, then the URLs can not be updated.

Before, the similar logic was fine because finally the URLs will be corrected by `updateUi`.

Reply:
Right, I forgot these buttons can conditionally be absent, will check.

Reply:
Fixed, I replicated the `updateUi` logic again.

Reply:
Hmm .... the code I wrote at that time was quite dirty. If there is a chance for me, I would write it like this:

(haven't tested ....)

```js
	// synchronously set clone button states and urls here to avoid flickering
	// on page load. initRepoCloneLink calls this when proto changes.
	(window.updateCloneStates = function() {
		const httpsBtn = document.getElementById('repo-clone-https');
		const sshBtn = document.getElementById('repo-clone-ssh');
		let isSSH = (localStorage.getItem('repo-clone-protocol') || 'https') === 'ssh' && sshBtn !== null;

		// there must be at least one clone button (by context/repo.go). if no ssh, then there must be https.
		let btn = isSSH ? sshBtn : httpsBtn;
		if (!btn) return; // the btn should be null, but if bugs occurs, should there be an Error?

		const link = btn.getAttribute('data-link');
		for (const el of document.getElementsByClassName('js-clone-url')) {
			el[el.nodeName === 'INPUT' ? 'value' : 'textContent'] = link;
		}

		if (httpsBtn) httpsBtn.classList[!isSSH ? 'add' : 'remove']('primary');
		if (sshBtn) sshBtn.classList[isSSH ? 'add' : 'remove']('primary');
	})();
```

Reply:
rewrote it, the crucial part is

```js
isSSH = value === 'ssh' && sshBtn || value !== 'ssh' && !httpsBtn
```

This will ensure it falls back to the other protocol when store indicates a value but the associated button is absent.